kyo-combinators/shared/src/main/scala/kyo/combinators.scala

```diff
@@ -0,0 +1,429 @@
+package kyo
+
+import kyo.*
+import kyo.Envs.HasEnvs
+import scala.annotation.implicitNotFound
+import scala.annotation.tailrec
+import scala.annotation.targetName
+import scala.concurrent.Future
+import scala.reflect.ClassTag
+import scala.util.NotGiven
+
+
+extension [A, S](effect: A < S)
+    @targetName("zipRight")
+    def *>[A1, S1](next: => A1 < S1): A1 < (S & S1) =
+        effect.map(_ => next)
+
+    @targetName("zipLeft")
+    def <*[A1, S1](next: => A1 < S1): A < (S & S1) =
+        effect.map(e => next.as(e))
+
+    @targetName("zip")
+    def <*>[A1, S1](next: => A1 < S1): (A, A1) < (S & S1) =
+        effect.map(e => next.map(n => (e, n)))
+
+    def as[A1, S1](value: => A1 < S1): A1 < (S & S1) =
```

Comment:
Can you inline the trivial methods? I think this will cause the trace to be captured in this file instead of in the caller's file.

Reply:
Good idea